ttnn/cpp/ttnn/operations/conv/conv2d/prepare_conv2d_weights.cpp

```diff
@@ -0,0 +1,380 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "conv2d.hpp"
+#include <sys/types.h>
+#include <cstdint>
+
+using namespace tt;
+namespace ttnn {
+namespace operations::conv {
+using sliding_window::SlidingWindowConfig;
+using sliding_window::ParallelConfig;
+
+namespace conv2d {
+
+void validate_weight_tensor(const ttnn::Tensor& weight_tensor) {
+    TT_ASSERT(!ttnn::has_storage_type_of(weight_tensor, ttnn::DEVICE_STORAGE_TYPE));
+    TT_ASSERT(weight_tensor.get_layout() == Layout::ROW_MAJOR);
+    TT_ASSERT(weight_tensor.get_shape().rank() == 4);
+}
+
+void validate_bias_tensor(const ttnn::Tensor& bias_tensor) {
+    TT_ASSERT(!ttnn::has_storage_type_of(bias_tensor, ttnn::DEVICE_STORAGE_TYPE));
+    TT_ASSERT(bias_tensor.get_shape().rank() == 4);
+    TT_ASSERT(bias_tensor.get_layout() == Layout::ROW_MAJOR);
```

Comment:
TT_ASSERT does not happen in release or on CI, I highlight recommend using TT_FATAL instead

Reply:
This code is copied from the current implementation, @mywoodstock @pavlejosipovic what do you think?